packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -140,98 +166,132 @@ export default class OctetstreamCodec implements ContentCodec {
                     : bytes.readUInt32LE(0);
 
             default: {
-                let result = 0;
-                let negative;
-
-                if (bigendian) {
-                    result = bytes.reduce((prev, curr) => prev << (8 + curr));
-                    negative = bytes.readInt8(0) < 0;
-                } else {
-                    result = bytes.reduceRight((prev, curr) => prev << (8 + curr));
-                    negative = bytes.readInt8(dataLength - 1) < 0;
-                }
-
-                if (signed && negative) {
-                    result -= 1 << (8 * dataLength);
-                }
-
+                const result = bigEndian
+                    ? signed
+                        ? bytes.readIntBE(0, dataLength / 8)
+                        : bytes.readUIntBE(0, dataLength / 8)
+                    : signed
+                    ? bytes.readIntLE(0, dataLength / 8)
+                    : bytes.readUIntLE(0, dataLength / 8);
                 // warn about numbers being too big to be represented as safe integers
                 if (!Number.isSafeInteger(result)) {
                     warn("Result is not a safe integer");
                 }
-
                 return result;
             }
         }
     }
 
-    private numberToValue(bytes: Buffer, options: { dataLength: number; bigendian: boolean }): number {
-        const { dataLength, bigendian } = options;
+    private numberToValue(bytes: Buffer, options: { dataLength: number; bigEndian: boolean }): number {
+        const { dataLength, bigEndian } = options;
         switch (dataLength) {
-            case 2:
-                return getFloat16(new DataView(bytes.buffer), bytes.byteOffset, !bigendian);
-            case 4:
-                return bigendian ? bytes.readFloatBE(0) : bytes.readFloatLE(0);
+            case 16:
+                return getFloat16(new DataView(bytes.buffer), bytes.byteOffset, !bigEndian);
+            case 32:
+                return bigEndian ? bytes.readFloatBE(0) : bytes.readFloatLE(0);
 
-            case 8:
-                return bigendian ? bytes.readDoubleBE(0) : bytes.readDoubleLE(0);
+            case 64:
+                return bigEndian ? bytes.readDoubleBE(0) : bytes.readDoubleLE(0);
 
             default:
-                throw new Error("Wrong buffer length for type 'number', must be 2, 4, 8, or is " + dataLength);
+                throw new Error("Wrong buffer length for type 'number', must be 16, 32, or 64 is " + dataLength);
         }
     }
 
+    private objectToValue(bytes: Buffer, schema?: DataSchema): DataSchemaValue {
+        if (schema?.type !== "object") {
+            throw new Error("Schema must be of type 'object'");
+        }
+
+        const result: { [key: string]: unknown } = {};
+        for (const propertyName in schema.properties) {
```

Comment:
to make this work the list of properties should always have the same order. Are we sure that this is consistent across different node instances? maybe it is safer to order them in alphabetical order?

Reply:
done :)

Reply:
I now updated this line to use `Object.getOwnPropertyNames`, which is ordered in Node version 6.4 and up, see https://node.green/#ES2015-misc-own-property-order